packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,360 @@
+export const isString = (s: unknown) =>
+  s instanceof String || typeof s === "string";
+
+function encodeDecodeObject(
+  obj: NestedObject,
+  { encode, depth = 0 }: EncodeDecodeObjectOptions,
+) {
+  const maxDepth = 5;
+
+  if (depth > maxDepth) {
+    throw new OrditSDKError("Object too deep");
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const key in obj) {
+    // eslint-disable-next-line no-prototype-builtins, no-continue
+    if (!obj.hasOwnProperty(key)) continue;
+
+    const value = obj[key];
+    if (isObject(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encodeDecodeObject(value as NestedObject, {
+        encode,
+        depth: depth + 1,
+      });
+    } else if (isString(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encode
+        ? encodeURIComponent(value as string)
+        : decodeURIComponent(value as string);
+    }
+  }
+
+  return obj;
+}
+
+// TODO: Make this function immutable
+/**
+ * Encodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_encodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: true });
+}
+
+// TODO: Make this function immutable
+/**
+ * Decodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_decodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: false });
+}
+
+// Temporary convertors until bignumber.js is integrated
+export function convertSatoshisToBTC(satoshis: number) {
+  return satoshis / 10 ** 8;
+}
+
+export function convertBTCToSatoshis(btc: number) {
+  return parseInt((btc * 10 ** 8).toString(), 10); // remove floating point overflow by parseInt
+}
+
+export function generateTxUniqueIdentifier(txId: string, index: number) {
+  return `${txId}:${index}`;
+}
+
+/**
+ * Converts an outpoint to id format.
+ *
+ * An outpoint refers to a specific output.
+ *
+ * Reference: https://developer.bitcoin.org/reference/transactions.html#outpoint-the-specific-part-of-a-specific-output
+ *
+ * @param outpoint Outpoint in string format `{txid}:{vout}`
+ * @returns id formatted string, `{txid}i{vout}`
+ */
+export function outpointToIdFormat(outpoint: string) {
+  if (outpoint.includes(":")) {
+    return outpoint.replace(":", "i");
+  }
+
+  return outpoint.includes("i") ? outpoint : `${outpoint}i0`;
+}
```

Comment:
not sure if the use of this function is safe, I think this fix was previously added due to the sudden trinity migration. 
I think we can throw error for those rpc that supposed to accept `id` but other format was passed, instead of transforming. 
Also because the assumption of `i0` is not always right

cc: @iamcrazycoder

Reply:
I agree w/ @kevzzsk. We need to refactor this and think of a better approach to fix this.

Reply:
Created a separate issue #47 since this changes the behaviour